common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -82,11 +85,13 @@ public void run() {
     private final Executor executor;
     private volatile boolean interrupted;
 
+    private final Lock processingLock = new ReentrantLock();
```

Comment:
I can see how this could be used to await till suspension complete - acutally when we're no longer running - but i don't see its usage yet - what's the purpose for it?

Reply:
This is used to ensure that even if we race and start a new Thread while the other is not done yet we never execute with more then one